persistence-modules/hibernate-queries-2/src/test/java/com/baeldung/hibernate/HibernateOnConflictUnitTest.java

```diff
@@ -0,0 +1,113 @@
+package com.baeldung.hibernate;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+
+import java.io.IOException;
+
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import com.baeldung.hibernate.pojo.Student;
+
+import jakarta.persistence.criteria.CriteriaBuilder;
+import jakarta.persistence.criteria.CriteriaQuery;
+import jakarta.persistence.criteria.Root;
+
+public class HibernateOnConflictUnitTest {
+
+    private static Session session;
+    private static Transaction transaction;
+
+    @BeforeClass
+    public static void setUp() throws IOException {
+        session = HibernateUtil.getSessionFactory()
+            .openSession();
+        transaction = session.beginTransaction();
+
+        session.createMutationQuery("delete from Student")
+            .executeUpdate();
+
+        Student student = new Student("John");
+        session.persist(student);
+
+        transaction.commit();
+        transaction = session.beginTransaction();
+    }
+
+    @AfterClass
+    public static void tearDown() {
+        transaction.rollback();
+        session.close();
+    }
+
+    @Test
+    public void givenInsertQueryWithOnConflictClause_whenNoConflict_ThenInsertNewRecord() {
+
+        long rowCountBefore = getRowCount();
+        int updated = session.createMutationQuery("""
+            insert into Student (studentId, name) values (2, 'Sean') 
+            on conflict(studentId) do update 
+            set name = excluded.name
+            """).executeUpdate();
+        long rowCountAfter = getRowCount();
+
+        assertEquals(updated, 1);
+        assertEquals(rowCountBefore, rowCountAfter);
```

Comment:
The argument order is reversed on all of the assertEquals calls. The expected value should be the first argument, and the actual value should be the second argument. Note that by using AssertJ rather than plain vanilla JUnit assertions, we reduce our chance of making this error.